src/test/test_runner.ts

```diff
@@ -6,6 +6,31 @@ import * as path from "path";
 import { isCI } from "../shared/constants";
 import { MultiReporter } from "./mocha_multi_reporter";
 
+/**
+ * Normalizes a filter path for cross-platform glob pattern matching.
+ * Handles Windows paths and extracts meaningful filter terms.
+ */
+function normalizeFilterPath(filter: string): string {
+	// Remove .test.js suffix if present
+	let cleanFilter = filter.replace(/\.test\.js$/, "");
+
+	// Convert backslashes to forward slashes for cross-platform compatibility
+	cleanFilter = cleanFilter.replace(/\\/g, "/");
+
+	// If it looks like a path, extract the last meaningful component
+	// e.g., "./src/test/dart/dartdoc" -> "dartdoc"
```

Comment:
We do not want this, we want to preserve paths, so that if we have multiple test files with the same name, we can distinguish them on the command line with the folders.

But also not that we might include `.test.ts` since we may auto-complete at the terminal, but the actual test is the compiled version that ends `.test.js` so we need to ensure both are handled. It might be simpler to filter first on the user query, and then afterwards on the `.test.js` suffix, because that would also work if the suffix was partially typed (like `dartdoc.test`).